forem/forem
# Point changes in customization and profile tabs to users_settings table

## Description
This PR serves as a "base" for the PRs I am creating that point changes to UsersSetting and UsersNotificationSetting attributes from `users` table to `users_settings` and `users_notification_settings` tables.
I am doing it this way to aid PR reviewers as much as possible.
Please review the Action plan below, and contact Arit if you need more clarification. Thank you for tackling this code review! ❤️ 

## ACTION PLAN (for [Step 5](https://github.com/forem/rfcs/blob/2dd7c4d32be2d72aca05389d53f614465ecff570/text/0000-user-settings.md) of the UsersSettings RFC)
1. Complete the implementation of the following tasks:
- Point changes on `brand_color1`, `brand_color2`, `config_theme`, `config_font`, `config_navbar`, `display_announcements`, `display_email_on_profile`, `display_sponsors`, `editor_version`, `experience_level`, `permit_adjacent_sponsors` from `users` table to `users_settings` table. [PR 13904](https://github.com/forem/forem/pull/13904) [this present PR] does this. ✅ 
- Point changes on `feed_fetched_at`, `feed_mark_canonical`, `feed_referential_link`, `feed_url` attributes from `users` table to `users_settings` table. [PR 13914](https://github.com/forem/forem/pull/13914). ✅ 
- Point changes on `inbox_type` and `inbox_guidelines` attributes from `users` table to `users_settings` table. [PR 13911](https://github.com/forem/forem/pull/13911). ✅ 
- Point changes on [all notification attributes](https://github.com/forem/forem/blob/ab6969b25c0dac27fe647566f7619ed220d70d4c/app/models/user.rb#L118) from `users` table to `users_settings` table. [PR 13910](https://github.com/forem/forem/pull/13910). ✅ 
- Delete the sync code that keeps the `users_settings` and `users_notification_settings` tables up-to-date with the values of all these attributes in `users` table. ✅   
2. Merge the PRs ~13910~, ~13911~ and ~13914~ into this present PR. ✅
3. Write new (and update existing) specs for the new functionality. _Note: I am choosing to write the specs altogether in order to avoid inevitable duplication of effort._ 😅 
4. Merge the consolidated PR 13904 into `main` after all CI tests pass.

## TO-DO list for this PR
- [x] Update relevant attributes in the `customization` and `profile` tabs to use the new `users_settings` table 
- [x] Ensure that the themes, fonts, etc. are applied when page is refreshed
- [X] Deleting a User should delete their associated records in `users_settings` and `users_notification_settings` tables
- [X] Go through the TO-DO's on the application 
- [X] Check for other instances where these fields are being used
- [ ] Fix all the tests and write new ones for the new controllers and concerns

## What type of PR is this? (check all applicable)



## QA Instructions, Screenshots, Recordings

1. While logged in to localhost, change the following settings and ensure they reflect in the app:
a. In the `settings/profile` tab, check/uncheck `Display email on profile` then check your profile to see if changes reflect.
b. Also within `settings/profile`, scroll down to `Branding` and choose two colors. Save, refresh and then check your profile to ensure that your banner color reflects the color you chose.
c. In the `settings/customization` tab, choose a combo of Site Theme, Base Reading Font and Site Navbar. Save, refresh, then ensure that these selections reflect in the app.
d. Also within `settings/customization`, switch your Editor Version, save, then Create a Post. You should see the editor version you chose.
e. To QA Display Announcements and Sponsors, go to `localhost:3000/admin` as an admin and create Announcements and Sponsors under the Advanced tab. Bear in mind that you can have just one sponsor per org_id. Next, within `settings/customization`, toggle Display Announcements and Sponsors and ensure these settings reflect in the app.
f. In the `settings/extensions` tab, scroll to the Publishing ... from RSS section. Enter `https://dev.to/feed` as your RSS Feed URL. Save, then refresh. A `Fetch Feed Now` button should now be visible within this section (if you previously had a Feed URL entered, this button was already visible). Click the button, then refresh your page. The Last Fetched value should be your current time.
2. For all other attributes (especially those in `notification_settings`), toggle them in the app, then in Rails Console, check that your user has the updated values in their `users_setting` and `users_notification_setting` records.

### UI accessibility concerns? NONE (backend changes)

## Added tests?

- [X] Not yet

## [Forem core team only] How will this change be communicated?
